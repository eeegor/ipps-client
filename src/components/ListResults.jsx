import React from 'react';
import './ListResults.scss';

export const ListResults = props => {
  const { currentCount, totalCount, currentPage, perPage } = props;
  const totalPages = Math.ceil(Math.round(currentCount / perPage));
  const numberPlaceholder = '---';
  console.log('total', totalPages);
  const formatNumber = value => parseInt(value, 10).toLocaleString();
  return (
    <div className="list-results">
      <h1 className="list-results__title">
        <b className="list-results__current-count">
          {!!currentCount ? formatNumber(currentCount) : numberPlaceholder}
        </b>
        {' / '}
        <b className="list-results__total-count">
          {!!totalCount ? formatNumber(totalCount) : numberPlaceholder}
        </b>{' '}
        Results
      </h1>
      <p className="list-results__info">
        <span className="info-block">
          <span className="list-results__page">Page:</span>
          <span className="list-results__pages">
            <b className="list-results__current-page">
              {(!!currentPage && formatNumber(currentPage)) ||
                numberPlaceholder}
            </b>
            {' / '}
            <b className="list-results__total-pages">
              {(!!currentCount && !!totalCount && formatNumber(totalPages)) ||
                numberPlaceholder}
            </b>
          </span>
        </span>
        <span className="info-block">
          <span>Per page:</span>
          <span>
            <b>{(!!perPage && formatNumber(perPage)) || numberPlaceholder}</b>
          </span>
        </span>
      </p>
    </div>
  );
};
