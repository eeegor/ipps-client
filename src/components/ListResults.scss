@import '../variables';

.list-results {
  height: 4rem;
  display: flex;
  flex-direction: column;
  padding: 0.75rem 1rem;
  background: #fff;
  border-bottom: 1px dashed rgba($color--primary, 0.3);

  @media (min-width: $size--tablet) {
    height: 5rem;
    padding: 0.9rem 1rem;
  }
}

.list-results__title {
  font-size: 0.7rem;
  line-height: 1.25rem;
  text-transform: uppercase;
  color: $color--primary;
  letter-spacing: 0.05rem;
  margin: 0;

  @media (min-width: $size--tablet) {
    font-size: 0.9rem;
    line-height: 1.5rem;
    margin-bottom: 0.5rem;
  }
}

.list-results__current-count {
  font-size: 1rem;

  @media (min-width: $size--tablet) {
    font-size: 1.25rem;
  }
}

.list-results__info {
  font-size: 0.8rem;
  line-height: 1.5;
  letter-spacing: 0.005rem;
  margin: 0;
  color: rgba(#333, 0.5);

  span {
    padding-right: 0.5rem;

    span {
      padding-right: 0.25rem;
    }
  }

  b {
    color: rgba(#333, 0.9);
  }
}
