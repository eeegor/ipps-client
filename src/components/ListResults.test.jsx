import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import { ListResults } from './ListResults';

describe('ListResults', () => {
	it('renders markup correctly', () => {
		const wrapper = <ListResults currentCount="20" totalCount="100" currentPage="2" perPage="5" />;
		const tree = renderer.create(wrapper).toJSON();
		expect(tree).toMatchSnapshot();
	});
	
  it('renders markup with placeholders correctly', () => {
    const wrapper = <ListResults>Hello</ListResults>;
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('renders markup with empty value correctly', () => {
    const wrapper = shallow(<ListResults currentCount="0" totalCount="100"  currentPage="1" perPage="5" />);
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render placeholders if no values', () => {
    const wrapper = shallow(<ListResults />);
    expect(wrapper.find('.list-results__current-count').contains('---')).toEqual(true);
    expect(wrapper.find('.list-results__total-count').contains('---')).toEqual(true);
    expect(wrapper.find('.list-results__current-page').contains('---')).toEqual(true);
    expect(wrapper.find('.list-results__total-pages').contains('---')).toEqual(true);
    expect(wrapper.find('.list-results__per-page').contains('---')).toEqual(true);
  });
  
  it('should render values correctly', () => {
    const wrapper = shallow(<ListResults currentCount="20" totalCount="100" currentPage="2" perPage="5" />);
    expect(wrapper.find('.list-results__current-count').contains('20')).toEqual(true);
    expect(wrapper.find('.list-results__total-count').contains('100')).toEqual(true);
    expect(wrapper.find('.list-results__current-page').contains('2')).toEqual(true);
    expect(wrapper.find('.list-results__total-pages').contains('4')).toEqual(true);
    expect(wrapper.find('.list-results__per-page').contains('5')).toEqual(true);
  });

  it('should render empty value correctly', () => {
    const wrapper = shallow(<ListResults currentCount="0" totalCount="100" currentPage="1" perPage="5" />);
    expect(wrapper.find('.list-results__current-count').contains('0')).toEqual(true);
    expect(wrapper.find('.list-results__total-count').contains('100')).toEqual(true);
    expect(wrapper.find('.list-results__current-page').contains('1')).toEqual(true);
    expect(wrapper.find('.list-results__total-pages').contains('1')).toEqual(true);
    expect(wrapper.find('.list-results__per-page').contains('5')).toEqual(true);
  });

});
