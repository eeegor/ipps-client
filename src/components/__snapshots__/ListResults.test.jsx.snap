// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListResults renders markup correctly 1`] = `
<div
  className="list-results"
>
  <h1
    className="list-results__title"
  >
    <b
      className="list-results__current-count"
    >
      20
    </b>
     / 
    <b
      className="list-results__total-count"
    >
      100
    </b>
     
    Results
  </h1>
  <p
    className="list-results__info"
  >
    <span
      className="info-block"
    >
      <span
        className="list-results__page"
      >
        Page:
      </span>
      <span
        className="list-results__pages"
      >
        <b
          className="list-results__current-page"
        >
          2
        </b>
         / 
        <b
          className="list-results__total-pages"
        >
          4
        </b>
      </span>
    </span>
    <span
      className="info-block"
    >
      <span>
        Per page:
      </span>
      <span>
        <b
          className="list-results__per-page"
        >
          5
        </b>
      </span>
    </span>
  </p>
</div>
`;

exports[`ListResults renders markup with empty value correctly 1`] = `
<div
  className="list-results"
>
  <h1
    className="list-results__title"
  >
    <b
      className="list-results__current-count"
    >
      0
    </b>
     / 
    <b
      className="list-results__total-count"
    >
      100
    </b>
     
    Results
  </h1>
  <p
    className="list-results__info"
  >
    <span
      className="info-block"
    >
      <span
        className="list-results__page"
      >
        Page:
      </span>
      <span
        className="list-results__pages"
      >
        <b
          className="list-results__current-page"
        >
          1
        </b>
         / 
        <b
          className="list-results__total-pages"
        >
          1
        </b>
      </span>
    </span>
    <span
      className="info-block"
    >
      <span>
        Per page:
      </span>
      <span>
        <b
          className="list-results__per-page"
        >
          5
        </b>
      </span>
    </span>
  </p>
</div>
`;

exports[`ListResults renders markup with placeholders correctly 1`] = `
<div
  className="list-results"
>
  <h1
    className="list-results__title"
  >
    <b
      className="list-results__current-count"
    >
      ---
    </b>
     / 
    <b
      className="list-results__total-count"
    >
      ---
    </b>
     
    Results
  </h1>
  <p
    className="list-results__info"
  >
    <span
      className="info-block"
    >
      <span
        className="list-results__page"
      >
        Page:
      </span>
      <span
        className="list-results__pages"
      >
        <b
          className="list-results__current-page"
        >
          ---
        </b>
         / 
        <b
          className="list-results__total-pages"
        >
          ---
        </b>
      </span>
    </span>
    <span
      className="info-block"
    >
      <span>
        Per page:
      </span>
      <span>
        <b
          className="list-results__per-page"
        >
          ---
        </b>
      </span>
    </span>
  </p>
</div>
`;
